File: 프로그래머스/Lv2/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    
    sort(people.begin(),people.end());
    if(people[0]>limit/2) return people.size();
    
    int temp=0;
    int Index=people.size()-1;
    while(temp<Index){
        
        if(people[temp]+people[Index]>limit){ //몸무게가 젤 큰 사람만 보내기
            Index--;
            answer++;
        }else{ //젤 큰 사람, 젤 작은 사람 동시에 보내기
            answer++;
            Index--;
            temp++;
        }
        if(temp==Index) answer++;
        
    }
    
    return answer;
}

File: 프로그래머스/Lv2/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort()
    i=0
    
    while(len(people)>i):
        if(len(people)==1):
            answer+=1
            people.pop()
        elif(people[i]+people[-1]<=limit):
            answer+=1
            people.pop()
            i+=1
        else:
            answer+=1
            people.pop()
            
    return answer
